src/Test.UnitTests.Sarif.Driver/Sdk/PluginDriverCommandTests.cs

```diff
@@ -60,26 +60,93 @@ public void PluginDriverCommand_ValidateInvocationPropertiesToLog_ShouldReturnFa
+
+            var mockFileSystem = new Mock<IFileSystem>();
+            mockFileSystem
+                .Setup(f => f.FileExists(It.IsAny<string>()))
+                .Returns(true);
+
+            var context = new TestAnalysisContext
+            {
+                PostUri = postUri,
+                OutputFilePath = null,
+                FileSystem = mockFileSystem.Object
+            };
+
+            PluginDriverCommand<AnalyzeOptionsBase>.SarifPost(context, httpClient);
+        }
+
+        [Fact]
+        public void PluginDriverCommand_NoExceptionIfPostUriNotPresent()
+        {
+            string outputFilePath = $"{Guid.NewGuid()}.txt";
+            using var httpClient = new HttpClientWrapper();
+
             var mockFileSystem = new Mock<IFileSystem>();
+            mockFileSystem
+                .Setup(f => f.FileExists(It.IsAny<string>()))
+                .Returns(true);
+
+            var context = new TestAnalysisContext
+            {
+                PostUri = null,
+                OutputFilePath = outputFilePath,
+                FileSystem = mockFileSystem.Object
+            };
+
+            PluginDriverCommand<AnalyzeOptionsBase>.SarifPost(context, httpClient);
+        }
 
-            Exception exception = await Record.ExceptionAsync(async () =>
+        [Fact]
+        public void PluginDriverCommand_ThrowsExitApplicationExceptionOnUnhandledException()
+        {
+            string postUri = "https://github.com/microsoft/sarif-sdk";
+            string outputFilePath = $"{Guid.NewGuid()}.txt";
+
+            var mockHttpClient = new Mock<HttpClientWrapper>();
+
+            mockHttpClient
+                .Setup(c => c.PostAsync(It.IsAny<string>(), It.IsAny<HttpContent>()))
+                .Throws(new InvalidOperationException());
+
+            var context = new TestAnalysisContext
             {
-                mockFileSystem
-                    .Setup(f => f.FileExists(It.IsAny<string>()))
-                    .Throws(new NullReferenceException());
-
-                await PluginDriverCommand<AnalyzeOptionsBase>.PostLogFile(postUri,
-                                                                          outputFilePath,
-                                                                          mockFileSystem.Object,
-                                                                          httpClient: null);
-            });
-
-            // TBD change this to an end-to-end test that injects an unhandled exception in post method.
-            exception.Should().BeOfType(typeof(ArgumentNullException));
+                PostUri = postUri,
+                OutputFilePath = outputFilePath,
+            };
+
+            Exception exception = Record.Exception(() =>
+                    PluginDriverCommand<AnalyzeOptionsBase>.SarifPost(context,
+                                                                      mockHttpClient.Object));
+
+            exception.Should().BeOfType(typeof(ArgumentException));
```

Comment:
>ArgumentException [](http://example.com/codeflow?start=47&length=17)

I changed this to make sure if all tests can pass, but will need to check if this is as expected, or a bug.